text_generation/causal_lm/cpp/continuous_batching/library/src/sampler.hpp

```diff
@@ -423,8 +424,9 @@ SamplerOutput Sampler::sample(std::vector<SequenceGroup::Ptr> & sequence_groups,
                 // in case of greedy search we always have a single parent sequence to sample from
                 running_sequences[0]->append_token(sampled_token_id, sequence_group_logits.data<const float>()[sampled_token_id]);
 
-                if (sampling_params.max_new_tokens == running_sequences[0]->get_generated_len() ||
-                    sampled_token_id == sampling_params.eos_token_id && !sampling_params.ignore_eos) {
+                if (sampling_params.min_new_tokens <= running_sequences[0]->get_generated_len() &&
+                    (sampling_params.max_new_tokens == running_sequences[0]->get_generated_len() ||
+                     sampled_token_id == sampling_params.eos_token_id && !sampling_params.ignore_eos)) {
```

Comment:
maybe we can add common `SequenceGroup::stop_generation()` with all such conditions?

Sequence group knows about:
- sampling params
- current length
- last/current token

And it can provide this answer